rust/protocol/src/storage/traits.rs

```diff
@@ -1,38 +1,72 @@
 //
-// Copyright 2020 Signal Messenger, LLC.
+// Copyright 2020-2022 Signal Messenger, LLC.
 // SPDX-License-Identifier: AGPL-3.0-only
 //
 
+//! Traits defining several stores used throughout the Signal Protocol.
+
 use async_trait::async_trait;
 use uuid::Uuid;
 
-use crate::state::{PreKeyId, SignedPreKeyId};
 use crate::{
-    IdentityKey, IdentityKeyPair, PreKeyRecord, ProtocolAddress, Result, SenderKeyRecord,
-    SessionRecord, SignedPreKeyRecord,
+    address::ProtocolAddress,
+    error::Result,
+    sender_keys::SenderKeyRecord,
+    state::{PreKeyId, PreKeyRecord, SessionRecord, SignedPreKeyId, SignedPreKeyRecord},
+    IdentityKey, IdentityKeyPair,
 };
 
+/// Handle to FFI-provided context object.
+///
+/// This object is not manipulated in Rust, but is instead passed back to each FFI
+/// method invocation. This argument should just be [None] for all clients of the Rust-only API.
 pub type Context = Option<*mut std::ffi::c_void>;
 
+/// Each Signal message can be considered to have exactly two participants, a sender and receiver.
+///
+/// [IdentityKeyStore::is_trusted_identity] uses this to ensure the identity provided is configured
+/// for the appropriate role.
+///
+/// *TODO: consider moving this enum into [crate::protocol]?*
```

Comment:
Hrm. Does not belong in a doc comment, but a good point. On the other hand, it really is a generic concept: it's mostly used here, but it's used for internal hygiene in the sealed sender implementation as well. Maybe it should just go in `util`.

Reply:
Moved into a non-doc comment and suggested a move to `utils.rs`!